src/Mod/Import/App/dxf/ImpExpDxf.cpp

```diff
@@ -103,7 +103,10 @@ bool ImpExpDxfRead::ReadEntitiesSection()
         // TODO: We do end-to-end joining or complete merging as selected by the options.
         for (auto& shapeSet : ShapesToCombine) {
             m_entityAttributes = shapeSet.first;
-            CombineShapes(shapeSet.second, "Compound");
+            CombineShapes(shapeSet.second,
+                          m_entityAttributes.m_Layer == nullptr
+                              ? "Compound"
+                              : m_entityAttributes.m_Layer->Name.c_str());
```

Comment:
```suggestion
            std::string compoundPrefix("Compound");
            CombineShapes(shapeSet.second,
                          m_entityAttributes.m_Layer == nullptr
                              ? compoundPrefix.c_str()
                              : compoundPrefix.append(m_entityAttributes.m_Layer->Name).c_str());
```
Suggestion to avoid the situation whereby when using `Make layers into blocks` and `Use layers` compounds from the "0" layer (or any other layer whose name starts with a digit) have the first digit replaced by an underscore. For the "0" layer this results in its first compound being named "_".

My understanding is that adding a letters-only prefix works around FreeCAD core doing the underscore replacement.

Apologies for my poor C++ coding skills.

![image](https://github.com/user-attachments/assets/a1504bf7-65c9-4d6b-95a6-f05949053fc5)
Screenshot with an import before and after this suggestion is applied

Reply:
I like this solution (Would even maybe add a _ between compound and the layer name?) but it's cosmetic. Want to add it @kpemartin ? Otherwise I just tested, this looks fine and ready to go.

Reply:
One thing the code could do to avoid the digits generated for uniqueness would be to include the attributes that separate the objects in each compound from those in other compounds in the same Layer, so one of the compounds in the original example would be "dashedred_red_dashed" (though that seems absurd in this example, people don't usually call the layers by their attributes...) thus the name is (almost) assuredly unique, and the (leading digit)->'_' would not occur.

But that sounds like a lot of trouble, though it would look pretty in a real-world import. It would also implicitly answer the question that is sure to pop up "Why didn't it merge everything into a single object?"

I think I'd prefer to have this spun off as a new Issue.